### PR TITLE
add debian support

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: templates/rhsm-katello-reconfigure.erb [hunk 1]
```diff
@@ -31,6 +31,17 @@ then
 
   # configure rhsm
   # the config command was introduced in rhsm 0.96.6
+  subscription-manager config \
```
Comment (sean797) [inline templates/rhsm-katello-reconfigure.erb]: can we DRY this? Maybe wrap just `BASEURL= foo` around an if else?

File: templates/rhsm-katello-reconfigure.erb [hunk 2]
```diff
@@ -31,6 +31,17 @@ then
...
+    --server.hostname="$KATELLO_SERVER" \
+    --server.prefix="$PREFIX" \
+    --server.port="$PORT" \
+    --rhsm.repo_ca_cert="%(ca_cert_dir)s$KATELLO_SERVER_CA_CERT" \
+    --rhsm.baseurl="$BASEURL"
+elif uname -v | grep -qi debian
```
Comment (sean797) [inline templates/rhsm-katello-reconfigure.erb]: Can we use /etc/os-release or a fact for this? Though I don't want to add another dependency.

File: templates/rhsm-katello-reconfigure.erb (hunk 1 above)
Comment (mdellweg) [reply templates/rhsm-katello-reconfigure.erb]: sure

File: templates/rhsm-katello-reconfigure.erb (hunk 2 above)
Comment (mdellweg) [reply templates/rhsm-katello-reconfigure.erb]: ATM we transform the built RPM into a DEB. That way we can install and run the script.

**Comment by mdellweg** [comment]
On a second thought, i'm not so sure, the DRY-ing was worth it, because there are some specialities going on around older rhel versions. I kept it on a separate commit to ease the comparison.

**Comment by sean797** [comment]
> ATM we transform the built RPM into a DEB. That way we can install and run the script.

Are you suggesting that is the official documented way for users to register Debian clients?

**Comment by mdellweg** [comment]
No, i'd prefer a proper debian package of course. Or considering, that this packet really only contains one script that is run on install, we could just run `/pub/katello-rhsm-consumer` from the katello-server directly.

**Comment by sean797** [comment]
Sorry I never got round to looking at this again earlier. Yes I agree, lets remove that second commit.

I think there is a discussion to have around how to we register clients, I'm thinking we should change to running `/pub/katello-rhsm-consumer` on a client and deprecate the RPM method. I think this means that we can stop using the dependence's and tooling we need to generate the RPM without adding more for generating an DEP and keeping the method for registering clients the same.

I'm not suggesting this PR deprecates RPM, just that I think the method on both systems should be the same, if not very similar.

@ehelms @jlsherrill @ekohl others; agree? disagree? any other views?

**Comment by ekohl** [comment]
I'm not that familiar with the subscription manager workflow.

**Comment by jlsherrill** [comment]
@sean797 I think what you said makes sense, I'm just not sure how users would react?  I *think* that if i were an admin i would prefer an rpm, but mainly because i tend to not want to run random scripts from a webserver.  With the rpm I can sign it with a gpg key from my company, i can't as easily do that with a random script. 

In reality I'm not sure if users *actually* care.  The script was added to support atomic hosts (who can't install an rpm), but it does make sense to try to reduce the bootstrap steps to a single set of instructions.

**Comment by ekohl** [comment]
> The RPM method, like you say, can be made more secure be signing the RPM, though I've never seen anyone actually do that.

Isn't that the goal of https://github.com/theforeman/puppet-certs/pull/188?

**Comment by PaulSD** [comment]
No, the goal of #188 is to facilitate signing/verifying the repo metadata that is generated by Pulp, not the RPMs themselves.  It is assumed that the RPMs served by Pulp will already be signed when Pulp retrieves them from the upstream repo ... So Pulp doesn't need to re-sign the RPMs, it only needs to sign the new metadata that Pulp itself generates.

In addition, #188 updates this bootstrap RPM so that it will install the GPG key generated by #188, so clients are unlikely to trust that GPG key until after this bootstrap RPM is installed, so it won't help to sign this bootstrap RPM with that key...

**Comment by ekohl** [comment]
@PaulSD thanks for explaining that distinction. I'll get back to reviewing it tomorrow.

**Comment by ekohl** [comment]
I'm leaning to including this in https://github.com/theforeman/puppet-certs/pull/192. @stbenjam could you have a look as well?

**Comment by ekohl** [comment]
Thanks!

**Pull request merged**